Windows/MainWindow.cpp

```diff
@@ -365,63 +360,60 @@ namespace MainWindow
 		ShowWindow(hwndMain, SW_MINIMIZE);
 	}
 
+	// Note that this also updates the config! Not very clean.
 	RECT DetermineWindowRectangle() {
-		RECT rc;
+		const int virtualScreenWidth = GetSystemMetrics(SM_CXVIRTUALSCREEN);
+		const int virtualScreenHeight = GetSystemMetrics(SM_CYVIRTUALSCREEN);
+		const int virtualScreenX = GetSystemMetrics(SM_XVIRTUALSCREEN);
+		const int virtualScreenY = GetSystemMetrics(SM_YVIRTUALSCREEN);
+		const int currentScreenWidth = GetSystemMetrics(SM_CXSCREEN);
+		const int currentScreenHeight = GetSystemMetrics(SM_CYSCREEN);
 
-		const int screenWidth = GetSystemMetrics(SM_CXVIRTUALSCREEN);
-		const int screenHeight = GetSystemMetrics(SM_CYVIRTUALSCREEN);
-		const int screenX = GetSystemMetrics(SM_XVIRTUALSCREEN);
-		const int screenY = GetSystemMetrics(SM_YVIRTUALSCREEN);
-
-		if (!g_Config.bFullScreen) {
-			bool visibleHorizontally = ((g_Config.iWindowX + g_Config.iWindowWidth) >= screenX) &&
-				((g_Config.iWindowX + g_Config.iWindowWidth) < (screenWidth + g_Config.iWindowWidth));
+		bool resetPositionX = true;
+		bool resetPositionY = true;
 
-			bool visibleVertically = ((g_Config.iWindowY + g_Config.iWindowHeight) >= screenY) &&
-				((g_Config.iWindowY + g_Config.iWindowHeight) < (screenHeight + g_Config.iWindowHeight));
+		if (g_Config.iWindowWidth > 0 && g_Config.iWindowHeight > 0 && !g_Config.bFullScreen) {
+			bool visibleHorizontally = ((g_Config.iWindowX + g_Config.iWindowWidth) >= virtualScreenX) &&
+				((g_Config.iWindowX + g_Config.iWindowWidth) < (virtualScreenWidth + g_Config.iWindowWidth));
 
-			if (!visibleHorizontally)
-				g_Config.iWindowX = -1;
+			bool visibleVertically = ((g_Config.iWindowY + g_Config.iWindowHeight) >= virtualScreenY) &&
+				((g_Config.iWindowY + g_Config.iWindowHeight) < (virtualScreenHeight + g_Config.iWindowHeight));
 
+			if (visibleHorizontally)
+				resetPositionX = false;
 			if (!visibleVertically)
```

Comment:
Should this be reversed - if it's visibleHorizontally reset X, visibileVertically, reset Y?  It seems like this resets the Y when it IS visible?

-[Unknown]